cilium-cli/cli/version.go

```diff
@@ -16,9 +17,13 @@ import (
 )
 
 func getLatestStableVersion() string {
-	resp, err := http.Get("https://raw.githubusercontent.com/cilium/cilium/main/stable.txt")
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+	const stableVersionURL = "https://raw.githubusercontent.com/cilium/cilium/main/stable.txt"
+	resp, err := client.Get(stableVersionURL)
 	if err != nil {
-		return "unknown"
+		return fmt.Sprintf("unknown (You can manually retrieve it by accessing %s)", stableVersionURL)
```

Comment:
We should output the error here.

Reply:
Yes, I agree. The getLatestStableVersion function can be updated to the following logic:
```go
const stableVersionDefaultRequestTimeout = 10 * time.Second

func getLatestStableVersion() (string, error) {
	client := &http.Client{
		Timeout: stableVersionDefaultRequestTimeout,
	}

	resp, err := client.Get("https://raw.githubusercontent.com/cilium/cilium/main/stable.txt")
	if err != nil {
		return "", errors.Wrap(err, "failed to fetch stable version")
	}
	defer resp.Body.Close()

	b, err := safeio.ReadAllLimit(resp.Body, safeio.KB)
	if err != nil {
		return "", errors.Wrap(err, "failed to read stable version")
	}

	return strings.TrimSpace(string(b)), nil
}
```

```go
stableVersion, err := getLatestStableVersion()
if err != nil {
	fmt.Printf("cilium image (running): unknown. Unable to obtain stable version. Reason: %s\n", err.Error())
} else {
	fmt.Printf("cilium image (stable): %s\n", stableVersion)
}
```